visualizations/frontend/filters/v2/TDatePicker.vue

```diff
@@ -0,0 +1,282 @@
+<template>
+    <div class="relative flex t-date-picker">
+        <!-- Date Picker -->
+        <base-date-picker
+            read-only
+            show-clear-button
+            confirm
+            confirm-text="Apply"
+            v-model="date"
+            ref="datePicker"
+            :range="mode === 'range'"
+            :format="dateFormat"
+            :range-separator="seperator"
+            @confirm="handleConfirm"
+            @pick="setCustomPreset"
+            @open="fetchLayerData"
+        >
+            <div class="presets-dropown" slot="sidebar">
+                <div
+                    v-for="preset in presets[mode]"
+                    :key="preset.key"
+                    :style="{ color: preset.key === selectedPreset.key ? '#1284e7' : '#73879c' }"
+                    class="cursor-pointer"
+                    @click="handlePreset(preset)"
+                >
+                    {{ preset.label }}
+                </div>
+            </div>
+
+            <div slot="icon-clear"></div>
+            <div slot="icon-calendar">
+                <div class="flex items-center gap-1">
+                    <reload-icon @click="handleReload" class="cursor-pointer" />
+                    <close-icon @click="handleClear" class="cursor-pointer" />
```

Comment:
Could we use a button here? This would make the element focusable and would remove the need for changing the cursor.